kdumpctl

```diff
@@ -1274,20 +1275,54 @@ do_estimate()
 	elif [[ ${baseline: -1} == "G" ]]; then
 		baseline=$((${baseline%G} * 1024))
 	elif [[ ${baseline: -1} == "T" ]]; then
-		baseline=$((${baseline%Y} * 1048576))
+		baseline=$((${baseline%T} * 1048576))
 	fi
 
 	# The default pre-reserved crashkernel value
 	baseline_size=$((baseline * size_mb))
 	# Current reserved crashkernel size
 	reserved_size=$(get_reserved_mem_size)
 	# A pre-estimated value for userspace usage and kernel
-	# runtime allocation, 64M should good for most cases
-	runtime_size=$((64 * size_mb))
+	# runtime allocation, 32M should good for most cases
+	runtime_size=$((32 * size_mb))
 	# Kernel image size
 	kernel_size=$(get_kernel_size "$KDUMP_KERNEL")
 	# Kdump initramfs size
 	initrd_size=$(du -b "$TARGET_INITRD" | awk '{print $1}')
+	# Get the decompressed size of initramfs contents
+	local _tmp_unpack_dir="$KDUMP_TMPDIR/initrd_unpack"
+	mkdir -p "$_tmp_unpack_dir"
+	trap 'rm -rf "$_tmp_unpack_dir"' EXIT
+
+	(cd "$_tmp_unpack_dir" && lsinitrd "$TARGET_INITRD" --unpack >/dev/null 2>&1)
+
+	unpacked_initramfs_size=0
+	# Add size of any squashfs/erofs images after unpacking/mounting
+	while read -r _img; do
+		local _img_mnt="$_tmp_unpack_dir/img_$$"
+		local _img_fs="squashfs"
+		mkdir -p "$_img_mnt"
+
+		if [[ "$_img" == *"erofs-root.img" ]]; then
+			_img_fs="erofs"
+			# lsinitrd use fsck.erofs to extract erofs-root.img.
+			if [[ -e "$_tmp_unpack_dir/erofs-root" ]]; then
+				unpacked_initramfs_size=$(du -sb "$_tmp_unpack_dir/erofs-root" | awk '{print $1}')
+			fi
+			if [[ "$unpacked_initramfs_size" -ne 0 ]]; then
+				break
+			fi
+		fi
+
+		if mount -t "$_img_fs" "$_img" "$_img_mnt" 2>/dev/null; then
+			unpacked_initramfs_size=$(du -sb "$_img_mnt" | awk '{print $1}')
+			umount "$_img_mnt"
+		else
+			derror "mount -t $_img_fs $_img failed!"
+			exit 1
+		fi
+	done < <(find "$_tmp_unpack_dir" -name "*.img")
```

Comment:
Maybe it's better to put the above into a separate function as there are already many lines of code do_estimate?